### PR TITLE
feat: clean your room

### PR DESCRIPTION
**Pull request by lelithium**

### CODE REVIEW

File: tool/runners/c.py [hunk 1]
```diff
@@ -31,3 +32,6 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 return RuntimeError(e)
+
+    def cleanup(self):
+        os.remove(self.executable)
```
Comment (badouralix) [inline tool/runners/c.py]: Newlines ftw

**Review by badouralix — APPROVED** [review]
:shipit:

File: tool/runners/c.py (hunk 1 above)
Comment (Ayc0) [inline tool/runners/c.py]:
```suggestion
        os.remove(self.executable)

```

File: tool/runners/cpp.py [hunk 2]
```diff
@@ -40,3 +41,6 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 return RuntimeError(e)
+
+    def cleanup(self):
+        os.remove(self.executable)
```
Comment (Ayc0) [inline tool/runners/cpp.py]:
```suggestion
        os.remove(self.executable)

```

File: tool/runners/go.py [hunk 3]
```diff
@@ -34,3 +34,6 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 raise RuntimeError(e)
+
+    def cleanup(self):
+        os.remove(self.executable)
```
Comment (Ayc0) [inline tool/runners/go.py]:
```suggestion
        os.remove(self.executable)

```

File: tool/runners/nim.py [hunk 4]
```diff
@@ -40,3 +41,6 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 return None
+
+    def cleanup(self):
+        os.remove(self.executable)
```
Comment (Ayc0) [inline tool/runners/nim.py]:
```suggestion
        os.remove(self.executable)

```

File: tool/runners/java.py [hunk 5]
```diff
@@ -34,3 +34,6 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 return RuntimeError(e)
+
+    def cleanup(self):
+        self.temporary_directory.cleanup()
```
Comment (Ayc0) [inline tool/runners/java.py]:
```suggestion
        self.temporary_directory.cleanup()

```

File: tool/runners/rust.py [hunk 6]
```diff
@@ -43,3 +43,7 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 raise RuntimeError(e)
+
+    def cleanup(self):
+        os.remove(self.executable)
+        self.tmpdir.cleanup()
```
Comment (Ayc0) [inline tool/runners/rust.py]:
```suggestion
        self.tmpdir.cleanup()

```

File: tool/runners/vlang.py [hunk 7]
```diff
@@ -31,3 +32,6 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 return RuntimeError(e)
+    
+    def cleanup(self):
+        os.remove(self.executable)
```
Comment (Ayc0) [inline tool/runners/vlang.py]:
```suggestion
        os.remove(self.executable)

```

File: tool/runners/zig.py [hunk 8]
```diff
@@ -39,3 +44,6 @@ def exec(self, input):
             else:
                 # subprocess exited with another error
                 return RuntimeError(e)
+    def cleanup(self):
+        self.build_dir.cleanup()
+        os.remove(self.executable)
```
Comment (Ayc0) [inline tool/runners/zig.py]:
```suggestion
        os.remove(self.executable)

```

**Comment by Ayc0** [comment]
oh yeah, j'ai causé un conflit \o/

**Pull request closed without merging**